app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show, :update, :destroy]
+
+  def index
+    @users = User.all
+    render json: @users
+  end
+
+  def show
+    render json: @user
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      logger.info "User created successfully: #{@user.inspect}"
+      redirect_to @user, notice: 'User was successfully created.'
+    else
+      logger.error "Failed to create user: #{@user.errors}"
+      render json: @user.errors, status: :unprocessable_entity
+    end
+  end
+  
+
```

Comment:
Add an `edit` method with its corresponding view (`form` should be a reusable partial)